Updated link, but also refreshed screenshots and cleaned up content. (#199)
mulesoft/docs-runtime-manager: modules/ROOT/pages/managing-schedules.adoc

endif::[]
:keywords: schedules, cloudhub, polling

> The *Schedules* tab in the left navigation menu of the CloudHub console enables you to view and control any polling elements within the flows of your deployed applications. On this tab, you can manage polling elements, change the polling frequency, or run them immediately – all during runtime, without making any changes in the underlying application.

Reviewer: ```suggestion
The *Schedules* tab in the left navigation menu of the CloudHub console enables you to view and control any polling elements within the flows of your deployed applications. On this tab, you can manage polling elements, change the polling frequency, or run polls immediately, all during runtime, without making any changes in the underlying application.
```

== Use Cases

The *Schedules* tab is useful in the following scenarios:

* You have a job that runs nightly but you want to debug it now.
+

> You can trigger it while it's running without updating the schedule in your underlying application.

Reviewer: ```suggestion
You can trigger the job while it's running without updating the schedule in your underlying application.
```

* You want to change the schedules in your application at runtime without redeploying.
* You want to disable a scheduled job while one of the applications you are connecting to is undergoing maintenance.
* You want runtime visibility into the schedules in your application.

> == Considerations

Reviewer: Be careful using this title to label a bunch of loosely related bits of information. This list looks pretty good. Sometimes writers use "Requirements for..." or "Restrictions to..." rather than "Considerations," which is a bit vague.

When using the *Schedules* tab, consider the following:

* Schedulers are managed by CloudHub and are triggered in only one worker even if more are available.
* Setting a polling schedule for less than a 10-second frequency between calls is not recommended.
* If a scheduled job is not triggered one or more times because the application is down, CloudHub triggers the job as soon as the application restarts.

> * CloudHub scheduler reads a schedule job configuration every time it runs.

Reviewer: Is this wording correct, "a schedule job configuration"? Is it accurate to use only "a job configuration"?

+

> If you update the schedule, the change is not immediate. The job executes as it was initially configured before the scheduler picks up the next setting for the next time it runs. To apply your changes immediately, run your jobs manually after updating the schedule.

Reviewer: Could we replace this para with only the final sentence and retain meaning?

* If poll execution time takes longer than 12 hours, the scheduler fails and is re-executed.

> * CloudHub schedules cannot be managed using scripts.

Reviewer: Intentional passive voice?

+
Use the CloudHub API or *Schedules* tab in Runtime Manager to manage CloudHub schedules.

> // Flow names can contain only letters (a-zA-Z), numbers (0-9), hyphen (-), underscore (_), and period (.).

Reviewer: ```suggestion
// Flow names can contain only letters (a-z and A-Z), numbers (0-9), hyphen (-), underscore (_), and period (.).
```

== View Application Schedules in the Console

> To use the scheduling feature in Runtime Manager, deploy an application that includes one or more polling elements. This example includes two simple flows, each with a polling scope.

Reviewer: Consider ending with a colon to point directly to a graphic element, and/or changing "This" to "The following."

----
=====

> CloudHub synchronizes your schedules automatically, overriding any synchronize property set on the flows containing the polling scopes, so you don't need to do any other special configuration.

Reviewer: ```suggestion
CloudHub synchronizes your schedules automatically, overriding any synchronize property set on the flows containing the polling scopes so that you don't need to perform any other special configuration.
```

After deploying an app, you can view the polling elements in Runtime Manager by selecting the app and clicking *Schedules* in the left navigation menu. Runtime Manager lists the polling elements in your application in the order they appear in your application:

image::ch-view-polling-schedule.png[View Schedules]

> The name of each element is determined by the `doc:name` attribute in the polling scopes, if specified in the XML. If you do not specify a `doc:name`, CloudHub applies the name, *yourFlowname Poll*.

Reviewer: ```suggestion
The name of each element is determined by the `doc:name` attribute in the polling scopes, if specified in the XML. If you do not specify `doc:name`, CloudHub applies the name *yourFlowname Poll*.
```

The *Last Run* column shows the date and time each polling element last ran. You can view the logs to see more information about when the jobs started and ended scheduled runs.

== Manage Schedules at Runtime

To manage schedules in Runtime Manager:

. Click *Applications* in the left pane, click the app, and then click *Schedules*:
+
image::ch-schedule-change.png[View Schedules]
+
. Select the elements to manage.

> * Click *Run now* to override the schedule and run the job immediately.

Reviewer: ```suggestion
. Click *Run now* to override the schedule and run the job immediately.
```

Reviewer: Here and below, STET * if these are choices. If they are chronological, they should be .

+

> This message appears: `The schedules have been queued to run.`

Reviewer: Do we need to document the success of the step?

Author: No.

> * Click *Disable* to suspend a polling element.

Reviewer: ```suggestion
. Click *Disable* to suspend a polling element.
```

. In the *Schedule* column, click the frequency of a polling element to open the schedule editor:
+
image::ch-set-poll-freq.png[Set Polling Frequency]
+
. In the schedule editor, change the polling frequency of the element or deselect *Enabled* to disable the polling schedule.
+ 
You can also <<Use Cron Expressions,use cron expressions>> to create more specific schedules, such as "every Thursday at 3:00 pm". 
. Click *Update* to save changed settings or click *Cancel* to exit without saving.
. Click *Logs* in the left navigation menu to verify that the job you selected ran as expected.

=== Use Cron Expressions

With cron expressions, you can create more complex polling frequencies, such as "every 5 minutes between 9:00 am and 10:00 on Fridays".

To use cron expressions to specify the polling frequency:

. In the schedule editor, click *Switch to advanced mode*.
. In the advanced mode page, click *Quick reference* to expand the table that lists required fields and allowed values:
+
image::ch-schedule-adv-mode-exp.png[Schedule Advanced Mode Expanded]
+
The cron expression in this example sets the polling schedule to every 10 seconds.
+
On this page:

> 

Reviewer: The following items are choices and not chrono?

Author: Yes.

> * Click *more* to view the http://www.quartz-scheduler.org/documentation[Quartz documentation] for information on building cron triggers.

Reviewer: ```suggestion
* Click *more* to view the http://www.quartz-scheduler.org/documentation[Quartz documentation] for information about building cron triggers.
```

+
Select a version, click *Documentation > User's Guide > Tutorial*, and then click *Lesson 6: CronTrigger*.
+

> All schedules are based in the UTC timezone, regardless of what region your CloudHub workers are on, so the timezone configuration is ignored.

Reviewer: ```suggestion
All schedules are based in the UTC timezone, regardless of what region your CloudHub workers are in, so the timezone configuration is ignored.
```
